### DISCUSSION TITLE
Exclusion / inclusion of disk drives and notes in the mapper settings

### DISCUSSION THREAD

**Discussion by SouceCalve** [SammyIAm/Moppy2#188] [Q&A]
I have a dire need to rule out which drives I can use with one controller, and also select the notes (range of notes) that can be played by it. The problem is that I don't understand how to do this, so please help.
I was able to fix 1-6 drives to mapper 1 using 
`(c+1)%7`
And my channel 7 is a speaker, to which I would like to output very high-frequency sounds, it is attached to mapper 2 using
`(c+1)/7`

**Comment by samphonic** [comment]
In the first script box (I'm fairly sure), you can use c<6 for the first 6,
and c==6 for the 7th one.  (If that doesn't work, let me know, not at a
computer right now, so apologies for the short response)

**Comment by SouceCalve** [comment]
For filtering devices work

**Comment by SouceCalve** [comment]
Not working for notes

**Comment by samphonic** [comment]
Ah, sorry, if I remember correctly, `n` is the variable for the note
number, so you could do e.g. `c<6 && n<48` (or whatever note range you
wanted).

I can't remember off hand if all 128 MIDI notes are supported out of the
box; you might need to modify a max note variable in the Arduino code
somewhere.  I can check once I'm at a computer again, but hopefully that's
enough to get your started.

*Edit: Double checked and `n` is the right variable, so I think that should work for you.*
